pkgs/development/python-modules/trainer/default.nix

```
@@ -30,6 +32,16 @@ buildPythonPackage {
     hash = "sha256-z6TOzWqE3NytkdG3nUzh9GpFVGQEXFyzSQ8gvdB4wiw=";
   };
 
+  patches = lib.optionals (pythonAtLeast "3.12") [
+    (fetchpatch {
+      name = "Add support for Python 3.12";
+      hash = "sha256-Yoory1fCwmczxNe7vknD9ynK/0ZtHuPQ2EiM1X9wKYM=";
+      url = "https://github.com/coqui-ai/Trainer/pull/143.patch";
```

```suggestion
      name = "add-support-for-python312.patch";
      hash = "sha256-Yoory1fCwmczxNe7vknD9ynK/0ZtHuPQ2EiM1X9wKYM=";
      url = "https://github.com/coqui-ai/Trainer/commit/0278012c7e6f5972b656d11757add4ab89f6d272.patch";
```
Is this commit what we need here?
Could you fetch a commit since the contents may be changed by a force push?

---

Yes, good catch! Resolved in latest.